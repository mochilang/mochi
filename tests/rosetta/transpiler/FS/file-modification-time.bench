{
  "duration_us": 308,
  "memory_bytes": 26368,
  "name": "main"
}