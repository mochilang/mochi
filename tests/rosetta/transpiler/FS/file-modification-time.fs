// Generated 2025-08-05 00:41 +0700

let mutable _nowSeed:int64 = 0L
let mutable _nowSeeded = false
let _initNow () =
    let s = System.Environment.GetEnvironmentVariable("MOCHI_NOW_SEED")
    if System.String.IsNullOrEmpty(s) |> not then
        match System.Int32.TryParse(s) with
        | true, v ->
            _nowSeed <- int64 v
            _nowSeeded <- true
        | _ -> ()
let _now () =
    if _nowSeeded then
        _nowSeed <- (_nowSeed * 1664525L + 1013904223L) % 2147483647L
        int _nowSeed
    else
        int (System.DateTime.UtcNow.Ticks % 2147483647L)

_initNow()
let _dictAdd<'K,'V when 'K : equality> (d:System.Collections.Generic.IDictionary<'K,'V>) (k:'K) (v:'V) =
    d.[k] <- v
    d
let _dictCreate<'K,'V when 'K : equality> (pairs:('K * 'V) list) : System.Collections.Generic.IDictionary<'K,'V> =
    let d = System.Collections.Generic.Dictionary<'K, 'V>()
    for (k, v) in pairs do
        d.[k] <- v
    upcast d
let _idx (arr:'a array) (i:int) : 'a =
    if i >= 0 && i < arr.Length then arr.[i] else Unchecked.defaultof<'a>
open System.Collections.Generic

open System

let mutable fs: System.Collections.Generic.IDictionary<string, int> = _dictCreate [("input.txt", 0)]
let rec main () =
    let mutable __ret : unit = Unchecked.defaultof<unit>
    try
        let __bench_start = _now()
        let __mem_start = System.GC.GetTotalMemory(true)
        let old: int = fs.[(string ("input.txt"))]
        printfn "%s" ("mod time was: " + (string (old)))
        let mutable mtime: int = _now()
        mtime <- int (_now())
        fs.["input.txt"] <- int mtime
        printfn "%s" ("mod time now: " + (string (mtime)))
        let __bench_end = _now()
        let __mem_end = System.GC.GetTotalMemory(true)
        printfn "{\n  \"duration_us\": %d,\n  \"memory_bytes\": %d,\n  \"name\": \"main\"\n}" ((__bench_end - __bench_start) / 1000) (__mem_end - __mem_start)

        __ret
    with
        | Return -> __ret
main()
