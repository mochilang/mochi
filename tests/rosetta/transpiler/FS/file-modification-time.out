mod time was: 0
mod time now: 985985230